packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/Drawing/helpers/withCoordsFromStream.js

```diff
@@ -0,0 +1,115 @@
+import { inject } from 'mobx-react'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+function storeMapper (stores) {
+  const {
+    eventStream
+  } = stores.classifierStore.drawing
+  return {
+    eventStream
+  }
+}
+
+function withCoordsFromStream (WrappedComponent) {
+  @inject(storeMapper)
+  class CoordsFromStream extends Component {
+    constructor () {
+      super()
+
+      this.state = {
+        activeCoordinates: []
+      }
+
+      this.firstSubscription = null
+      this.secondSubscription = null
+
+      this.finishDrawing = this.finishDrawing.bind(this)
+      this.startDrawing = this.startDrawing.bind(this)
+    }
+
+    componentDidMount () {
+      this.startDrawing()
+    }
+
+    componentWillUnmount () {
+      if (this.firstSubscription) {
+        this.firstSubscription.unsubscribe()
+      }
+      if (this.secondSubscription) {
+        this.secondSubscription.unsubscribe()
+      }
+    }
+
+    startDrawing () {
+      if (this.props.active) {
+        const { eventStream } = this.props
+        this.firstSubscription = eventStream.subscribe(event => {
+          if (event.type === 'pointerdown') {
+            this.secondSubscription = eventStream.subscribe(event => {
```

Comment:
I think something like this? https://medium.com/@paynoattn/3-common-mistakes-i-see-people-use-in-rx-and-the-observable-pattern-ba55fee3d031#ad2b

Rather than creating new streams, using `switchMap` on the existing stream to get what we want